src/osd/PrimaryLogPG.h

```diff
@@ -1504,8 +1504,74 @@ class PrimaryLogPG : public PG, public PGBackend::Listener {
   void snap_trimmer(epoch_t e) override;
   void kick_snap_trim() override;
   void snap_trimmer_scrub_complete() override;
+
+  bool _do_osd_op_pre(OpContext *ctx, OSDOp& osd_op, int& result);
```

Comment:
nit: the result args for _do_osd_op_{pre,post} would be clearer if they were pointers rather than references